src/impl/collection_notifier.hpp

```diff
@@ -194,8 +196,10 @@ class CollectionNotifier {
     void set_table(Table const& table);
     std::unique_lock<std::mutex> lock_target();
     SharedGroup& source_shared_group();
-
     std::function<bool (size_t)> get_modification_checker(TransactionChangeInfo const&, Table const&);
+    bool m_partial_sync_enabled = true;
+    std::shared_ptr<Realm> m_realm;
```

Comment:
Moving this declaration here is not valid (it needs appear after the mutex guarding it, so that it is destroyed before that mutex).